pkg/validator/promql_expression.go

```diff
@@ -170,6 +171,69 @@ func (h expressionUsesOnlyAllowedLabelsForMetricRegexp) Validate(_ unmarshaler.R
 	return errs
 }
 
+type expressionUsesOnlyAllowedLabelValuesForMetricRegexp struct {
+	allowedLabelValues map[string][]string
+	metricNameRegexp   *regexp.Regexp
+}
+
+func newExpressionUsesOnlyAllowedLabelValuesForMetricRegexp(paramsConfig yaml.Node) (Validator, error) {
+	params := struct {
+		allowedLabelValues map[string][]string `yaml:"allowedLabelValues"`
+		MetricNameRegexp   string              `yaml:"metricNameRegexp"`
+	}{}
+	if err := paramsConfig.Decode(&params); err != nil {
+		return nil, err
+	}
+	if len(params.allowedLabelValues) == 0 {
+		return nil, fmt.Errorf("missing allowed label values")
+	}
+	compiled, err := compileAnchoredRegexp(params.MetricNameRegexp)
```

Comment:
Not sure how we handle empty/not-set regexp in other validators, but I'd expect it to be either required or default to `.*` :thinking: 

TBH IDK if the unmarshalling is strict or not

Reply:
I changed it to default to `.*`, if no regexp is specified in params.

Will create a separate PR to make it a global default.

Reply:
Thanks!